crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -1300,6 +1350,25 @@ impl CheckpointBuilder {
                 None
             };
 
+            let version_specific_data = if self
+                .epoch_store
+                .protocol_config()
+                .checkpoint_summary_version_specific_data()
+            {
+                let matching_randomness_rounds: Vec<_> = effects
+                    .iter()
+                    .filter_map(|e| randomness_rounds.get(e.transaction_digest()))
+                    .copied()
+                    .collect();
+                Some(CheckpointVersionSpecificData::V1(
+                    CheckpointVersionSpecificDataV1 {
+                        randomness_rounds: matching_randomness_rounds,
+                    },
+                ))
+            } else {
+                None
+            };
```

Comment:
i think this could be moved into the CheckpointSummary constructor. The randomness rounds can be passed in unconditionally to keep the code clean, and ignored if the protocol version does not support storing them.

Reply:
done